### PR TITLE
Fix broken UDIMs in VP2 and add unit test

### LINKED ISSUE #3036: UDIM does not work on 0.23 plugin
Hello,
UDIM did not work when I used version 0.23 USD to create a preview shader in Maya2024. 
The result is as follows
Snipaste_2023-04-26_15-00-22
It can be seen that the texture is disordered
But when I used the 0.20 version of USD, the same USD file displayed correctly, as follows
21244fee016df3867d910a133c7d255d
f8658bfb74c5de87db8563eeb549bb7c
This is a simple test file

[test.zip](https://github.com/Autodesk/maya-usd/files/11329960/test.zip)

### PR DESCRIPTION
**Pull request by JGamache-autodesk**
Fixes #3036 and #3016 and adds MaterialX and UsdPreviewSurface rendering unit tests

### CODE REVIEW

**Comment by JGamache-autodesk** [comment]
@vlasovi There is a direct merge conflict with your #3043 that will have to be resolved by whoever merge lasts. I will wait until you merge your PR.

File: test/testSamples/MaterialX/grid_with_udims.usda [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    )
+    point3f[] points = [(-0.5, 0, 0.5), (-0.16666666, 0, 0.5), (0.16666669, 0, 0.5), (0.5, 0, 0.5), (-0.5, 0, 0.16666666), (-0.16666666, 0, 0.16666666), (0.16666669, 0, 0.16666666), (0.5, 0, 0.16666666), (-0.5, 0, -0.16666669), (-0.16666666, 0, -0.16666669), (0.16666669, 0, -0.16666669), (0.5, 0, -0.16666669), (-0.5, 0, -0.5), (-0.16666666, 0, -0.5), (0.16666669, 0, -0.5), (0.5, 0, -0.5)]
+    color3f[] primvars:displayColor = [(0.13320851, 0.13320851, 0.13320851)] (
+        customData = {
+            dictionary Maya = {
+                bool generated = 1
+            }
+        }
+    )
+    texCoord2f[] primvars:st = [(0.1, 0.1), (0.9, 0.1), (1.9, 0.1), (2.9, 0.1), (3.1, 0.1), (4.1, 0.1), (5.1, 0.1), (2.9, 0.9), (9.1, 0.1), (1.1, 2.1), (5.1, 5.1), (5.9, 0.9), (9.1, 0.9), (1.1, 2.9), (5.1, 5.9), (5.9, 5.9), (9.9, 0.9), (4.1, 0.9), (3.9, 0.9), (9.9, 0.1), (3.1, 0.9), (1.9, 2.9), (5.9, 5.1), (5.1, 0.9), (4.9, 0.9), (1.9, 2.1), (2.1, 0.9), (1.9, 0.9), (4.9, 0.1), (5.9, 0.1), (1.1, 0.9), (0.9, 0.9), (3.9, 0.1), (0.1, 0.9), (2.1, 0.1), (1.1, 0.1)] (
```
Comment (JGamache-autodesk) [inline test/testSamples/MaterialX/grid_with_udims.usda:29]: Notice there are gaps in the UDIMS, we jump directly from tile 1006 to tile 1010 and 1022. This makes sure gaps are well handled, and that UDIMs at higher V values are working.

The UDIM atlas in this case will be 10x3 with 8 tiles loaded and 28 empty tiles. Not efficient, but that is how the current codebase loads UDIMs.

File: lib/mayaUsd/render/vp2RenderDelegate/material.cpp [hunk 2]
```diff
@@ -3263,6 +3267,7 @@ void HdVP2Material::CompiledNetwork::_UpdateShaderInstance(
                         paramName = nodeName + "stOffset";
 #endif
                         status = _surfaceShader->setParameter(paramName, info._stOffset.data());
+                        updatedAttributes.insert(paramName);
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/render/vp2RenderDelegate/material.cpp:3270]: Will require the same for the new `cullStyle` parameter introduced in #3043

File: lib/mayaUsd/render/vp2RenderDelegate/material.cpp [hunk 3]
```diff
@@ -3161,11 +3161,13 @@ void HdVP2Material::CompiledNetwork::_UpdateShaderInstance(
                 if (isMaterialXNode) {
                     const MString paramName = "_" + nodeName + "file_sampler";
                     samplerStatus = _surfaceShader->setParameter(paramName, *sampler);
+                    updatedAttributes.insert(paramName);
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/render/vp2RenderDelegate/material.cpp:3164]: PR #3043 modifies all preceding lines, so automatic merge conflicts are to be expected.

**Review by vlasovi — DISMISSED** [review]

**Review by vlasovi — APPROVED** [review]

**Pull request merged**